pydantic/fields.py

```diff
@@ -825,3 +814,28 @@ def __eq__(self, other: Any) -> bool:
             other.default,
             other.default_factory,
         )
+
+
+def PrivateAttr(
+    default: Any = Undefined,
+    *,
+    default_factory: Optional[NoArgAnyCallable] = None,
+) -> Any:
+    """
+    Indicates that attribute is only used internally and never mixed with regular fields.
+
+    Types or values of private attrs are not checked by pydantic and it's up to you to keep them relevant.
+
+    Private attrs are stored in model __slots__.
+
+    :param default: the attribute’s default value
+    :param default_factory: callable that will be called when a default value is needed for this attribute
+      If both `default` and `default_factory` are set, an error is raised.
+    """
+    if default is not Undefined and default_factory is not None:
+        raise TypeError('cannot specify both default and default_factory')
```

Comment:
```suggestion
        raise ValueError('cannot specify both default and default_factory')
```
to be consistent with `ModelField`

Reply:
Good catch! I have committed this change.

Reply:
Humm, not sure I agree with this. `TypeError` is what gets raised when you call a function with the wrong arguments. I think better to be correct than consistent here (we could change `ModelField` in v2).

But I don't mind that much.

Reply:
Ah yes you're right so it was my mistake when I raised the `ValueError` in `ModelField` oops 😬

Reply:
Let's keep this for now and fix in both places later #2059